xknxproject/loader/application_program_loader.py

```diff
@@ -1,34 +1,32 @@
 """Application Program Loader."""
-from pathlib import Path
 from typing import Any
+from zipfile import Path
 
-from lxml.etree import iterparse  # pylint: disable=no-name-in-module
+from lxml import etree
 
 from xknxproject.models import ComObject, DeviceInstance
+from xknxproject.util import parse_dpt_types
 
-from ..util import parse_dpt_types
-from .loader import XMLLoader
 
-
-class ApplicationProgramLoader(XMLLoader):
+class ApplicationProgramLoader:
     """Load the application program from KNX XML."""
 
     def __init__(self, devices: list[DeviceInstance]):
         """Initialize the ApplicationProgramLoader."""
         self.devices = devices
 
-    async def load(self, extraction_path: Path) -> list[Any]:
+    def load(self, project_root_path: Path) -> list[Any]:
         """Load Hardware mappings and assign to devices."""
         application_programs: dict[
             str, list[DeviceInstance]
         ] = self._get_optimized_application_program_struct()
-        for application_program, devices in application_programs.items():
+        for application_program_file, devices in application_programs.items():
             com_object_mapping: dict[str, dict[str, str]] = {}
             com_objects: dict[str, ComObject] = {}
-            with open(
-                extraction_path / application_program, mode="rb"
+            with (project_root_path / application_program_file).open(
+                mode="rb"
             ) as application_xml:
-                for _, elem in iterparse(application_xml):
+                for elem in etree.parse(application_xml).iter():
```

Comment:
```suggestion
                for _, elem in iterparse(application_xml):
```

We should use `iterparse` here to avoid loading the whole XML into RAM.